nakama/socket.lua

```diff
@@ -0,0 +1,682 @@
+		channel_join = {
+			target = target,
+			type = type,
+			persistence = persistence,
+			hidden = hidden,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- send_channel_leave
+-- @param socket
+-- @param channel_id
+-- @param callback
+function M.send_channel_leave(socket, channel_id, callback)
+	assert(socket)
+	assert(_G.type(channel_id) == 'string')
+	local message = {
+		channel_leave = {
+			channel_id = channel_id,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- send_channel_message_send
+-- @param socket
+-- @param channel_id
+-- @param content
+-- @param callback
+function M.send_channel_message_send(socket, channel_id, content, callback)
+	assert(socket)
+	assert(_G.type(channel_id) == 'string')
+	assert(_G.type(content) == 'string')
+	local message = {
+		channel_message_send = {
+			channel_id = channel_id,
+			content = content,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- send_channel_message_remove
+-- @param socket
+-- @param channel_id
+-- @param message_id
+-- @param callback
+function M.send_channel_message_remove(socket, channel_id, message_id, callback)
+	assert(socket)
+	assert(_G.type(channel_id) == 'string')
+	assert(_G.type(message_id) == 'string')
+	local message = {
+		channel_message_remove = {
+			channel_id = channel_id,
+			message_id = message_id,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- send_match_data
+-- @param socket
+-- @param match_id
+-- @param presence
+-- @param op_code
+-- @param data
+-- @param reliable
+-- @param callback
+function M.send_match_data(socket, match_id, presence, op_code, data, reliable, callback)
+	assert(socket)
+	assert(_G.type(match_id) == 'string')
+	assert(_G.type(presence) == 'table')
+	assert(_G.type(op_code) == 'number')
+	assert(_G.type(data) == 'string')
+	assert(_G.type(reliable) == 'boolean')
+	local message = {
+		match_data = {
+			match_id = match_id,
+			presence = presence,
+			op_code = op_code,
+			data = data,
+			reliable = reliable,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- send_match_data_send
+-- @param socket
+-- @param match_id
+-- @param op_code
+-- @param data
+-- @param presences
+-- @param reliable
+-- @param callback
+function M.send_match_data_send(socket, match_id, op_code, data, presences, reliable, callback)
+	assert(socket)
+	assert(_G.type(match_id) == 'string')
+	assert(_G.type(op_code) == 'number')
+	assert(_G.type(data) == 'string')
+	assert(_G.type(presences) == 'table')
+	assert(_G.type(reliable) == 'boolean')
+	local message = {
+		match_data_send = {
+			match_id = match_id,
+			op_code = op_code,
+			data = data,
+			presences = presences,
+			reliable = reliable,
+		}
+	}
+	return socket_send(socket, message, callback)
+end
+
+--- send_match_join
+-- @param socket
+-- @param match_id
+-- @param token
+-- @param metadata
+-- @param callback
+function M.send_match_join(socket, match_id, token, metadata, callback)
```

Comment:
I think it would be great to rename some of these functions to remove the `"send"` part. e.g. here it would be `socket.join_match`. See function names in the .NET SDK for reference. https://github.com/heroiclabs/nakama-dotnet/blob/7d855ad80e5f69eb4c2841b89e9cd2dd28248385/Nakama/Socket.cs

Reply:
Done